File: backend/domain/entities/role.py
from dataclasses import dataclass, field
from datetime import datetime
from typing import Optional
import uuid


@dataclass
class Role:
    id: uuid.UUID
    name: str
    description: Optional[str] = None
    active: bool = True
    created_at: datetime = field(default_factory=datetime.utcnow)

    @classmethod
    def create(cls, name: str, description: Optional[str] = None) -> "Role":
        return cls(id=uuid.uuid4(), name=name, description=description)


@dataclass
class Permission:
    id: uuid.UUID
    module: str
    action: str
    description: Optional[str] = None

    @classmethod
    def create(cls, module: str, action: str, description: Optional[str] = None) -> "Permission":
        return cls(id=uuid.uuid4(), module=module, action=action, description=description)

File: frontend/src/types/role.ts
import type { Permission } from './api'

export interface PermissionRef {
  id: string
  module: string
  action: string
}

export interface RoleDetail {
  id: string
  name: string
  description: string | null
  active: boolean
  permissions: PermissionRef[]
  created_at: string
}

export interface RoleFormData {
  name: string
  description?: string
}

export interface RolePermissionsUpdate {
  permission_ids: string[]
}

export interface PermissionCatalogItem {
  id: string
  module: string
  action: string
  description: string | null
}

export interface PermissionFormData {
  module: string
  action: string
  description?: string
}

export type { Permission }
